### PR TITLE
Password Encryption

### PR DESCRIPTION
**Pull request by Ray-R4y**
# Description
Added password encryption via public key, and updated some checks due to the encrypted password
the database will now store the encrypted password

Fixes # (issue)
#65 
Fixed the issue where users would register with a pre-existing username , made username distinct and only one username can be stored in the database

### CODE REVIEW

File: src/public/login.cjs [hunk 1]
```diff
@@ -66,3 +94,22 @@ function onError () {
   // Let's tell the user that something wrong happened.
   document.getElementById('output').innerHTML = 'Status: Error communicating with server.'
 }
+
+// function encryptWithPublicKey(password){
```
Comment (RogerDMK) [inline src/public/login.cjs:98]: Think you just forgot to remove this

File: src/services/loginDb.cjs [hunk 2]
```diff
@@ -60,21 +53,47 @@ async function registerUser(username, password) {
...
+          }
         }
-      ).catch(reject) // TODO: find out why this is causing the server to crash
-    ).catch(reject) // TODO: find out why this is causing the server to crash
+      ).catch(reject)
+    ).catch(reject)
   })
 }
+
+// function decryptMessage(encrypted){
```
Comment (RogerDMK) [inline src/services/loginDb.cjs:90]: Think you forgot to remove this

**Review by RogerDMK — COMMENTED** [review]
The comments given should be addressed and tests need to be added, otherwise everything looks good. Style fits the general style employed by the group. Checks have been given for edge cases and encryption method has been changed to hashing.

File: src/public/login.cjs [hunk 3]
```diff
@@ -2,33 +2,64 @@
...
+      const password = document.getElementById('password').value
+      const encryptPassword = encryptWithPublicKey(password)
+      // send username and password to the server via json
+      request.send(JSON.stringify({ usernameInput: username, passwordInput: encryptPassword }))
+      // wait for server to respond back
+      request.addEventListener('load', receivedValue)
+    }
   })
 
   document.getElementById('registerButton').addEventListener('click', function (evt) {
```
[the remark this answers is not in the document]
Comment (Ray-R4y) [reply src/public/login.cjs:39]: refractored code to reduced code complexity and duplication

**Review by RogerDMK — APPROVED** [review]
All changes look good, Can be merged into main.
